src/script/E2EIdentity/E2EIdentity.ts

```diff
@@ -0,0 +1,275 @@
+ *
+ */
+
+import {container} from 'tsyringe';
+
+import {PrimaryModal, removeCurrentModal} from 'Components/Modals/PrimaryModal';
+import {Config} from 'src/script/Config';
+import {Core} from 'src/script/service/CoreSingleton';
+import {UserState} from 'src/script/user/UserState';
+import {removeUrlParameters} from 'Util/UrlUtil';
+import {supportsMLS} from 'Util/util';
+
+import {DelayTimerService} from './DelayTimer/DelayTimer';
+import {getModalOptions, ModalType} from './Modals';
+import {getOIDCServiceInstance} from './OIDCService';
+import {OIDCServiceStore} from './OIDCService/OIDCServiceStorage';
+
+export enum E2EIHandlerStep {
+  UNINITIALIZED = 'uninitialized',
+  INITIALIZED = 'initialized',
+  ENROLL = 'enroll',
+  SUCCESS = 'success',
+  ERROR = 'error',
+  SNOOZE = 'snooze',
+}
+
+interface E2EIHandlerParams {
+  discoveryUrl: string;
+  gracePeriodInMS: number;
+}
+
+class E2EIHandler {
+  private static instance: E2EIHandler | null = null;
+  private readonly core = container.resolve(Core);
+  private readonly userState = container.resolve(UserState);
+  private timer: DelayTimerService;
+  private discoveryUrl: string;
+  private gracePeriodInMS: number;
+  private currentStep: E2EIHandlerStep | null = E2EIHandlerStep.UNINITIALIZED;
+
+  private constructor({discoveryUrl, gracePeriodInMS}: E2EIHandlerParams) {
+    // ToDo: Do these values need to te able to be updated? Should we use a singleton with update fn?
+    this.discoveryUrl = discoveryUrl;
+    this.gracePeriodInMS = gracePeriodInMS;
+    this.timer = DelayTimerService.getInstance({
+      gracePeriodInMS,
+      gracePeriodExpiredCallback: () => null,
+      delayPeriodExpiredCallback: () => null,
+    });
+  }
+
+  /**
+   * Get the singleton instance of GracePeriodTimer or create a new one
+   * For the first time, params are required to create the instance
+   * After that, params are optional and can be used to update the grace period timer
+   * @param params The params to create the grace period timer
+   * @returns The singleton instance of GracePeriodTimer
+   */
+  public static getInstance(params?: E2EIHandlerParams) {
+    if (!E2EIHandler.instance) {
+      if (!params) {
+        throw new Error('GracePeriodTimer is not initialized. Please call getInstance with params.');
+      }
+      E2EIHandler.instance = new E2EIHandler(params);
+    }
+    return E2EIHandler.instance;
+  }
+
+  /**
+   * Reset the instance
+   */
+  public static resetInstance() {
+    E2EIHandler.instance = null;
+  }
+
+  /**
+   * @param E2EIHandlerParams The params to create the grace period timer
+   */
+  public updateParams({gracePeriodInMS, discoveryUrl}: E2EIHandlerParams) {
+    this.gracePeriodInMS = gracePeriodInMS;
+    this.discoveryUrl = discoveryUrl;
+    this.timer = DelayTimerService.getInstance({
+      gracePeriodInMS,
+      gracePeriodExpiredCallback: () => null,
+      delayPeriodExpiredCallback: () => null,
+    });
```

Comment:
This should probably call `updateParams` instead of `getInstance` since `getInstance` doesn't seem to update the params if it's already running